test/srt/cpu/test_moe_topk_softmax.py

```diff
@@ -0,0 +1,57 @@
+import itertools
+
+import pytest
+import sgl_kernel
+import torch
+
+torch.manual_seed(1998)
+
+
+@pytest.mark.parametrize(
+    "num_tokens, num_experts, topk, device",
+    list(
+        itertools.product(
+            [1, 16, 128, 512, 1024, 2048],  # num_tokens
+            [4, 8, 16, 32, 64, 128, 256],  # num_experts
+            [1, 2, 4],  # topk
+            ["cpu"],
+        )
+    ),
+)
+def test_topk_softmax(num_tokens, num_experts, topk, device):
+    test_dtype = torch.bfloat16
+
+    gating_output = torch.randn(
+        (num_tokens, num_experts), dtype=test_dtype, device=device
+    )
+    hidden_states = torch.randn(
+        (num_tokens, 1024), dtype=gating_output.dtype, device=device
+    )
+
+    topk_weights, topk_indices = torch.ops.sgl_kernel.topk_softmax_cpu(
+        hidden_states,
+        gating_output,
+        topk,
+        False,
+    )
+
+    # Native torch implementation
+    softmax_output = torch.softmax(gating_output, dim=-1)
+    topk_weights_ref, topk_indices_ref = torch.topk(softmax_output, topk, dim=-1)
+
+    # Verify the top-k weights and indices match the torch native ones
+    assert torch.allclose(
+        topk_weights_ref.to(test_dtype),
+        topk_weights.to(test_dtype),
+    ), f"Weights mismatch: torch={topk_weights_ref} vs SGLang={topk_weights}"
+
+    assert torch.allclose(
+        topk_indices_ref.sort()[0].to(torch.int32)[0, :],
+        topk_indices.sort()[0].to(torch.int32)[0, :],
+        atol=0,
+        rtol=0,
```

Comment:
scatter back topk_weights with regard to topk_indices and compare, take test_topk.py as a reference.

we may have identical values in topk_weights and different topk_indices.